implementation/src/main/java/io/smallrye/config/SmallRyeConfigBuilder.java

```diff
@@ -118,6 +119,11 @@ public SmallRyeConfigBuilder withSources(ConfigSource... configSources) {
         return this;
     }
 
+    public SmallRyeConfigBuilder withSources(Collection<ConfigSource> configSources) {
+        sources.addAll(configSources);
```

Comment:
What's the difference between this, and the other method that uses `Collections.addAll(sources, configSources)`?

Wondering if it makes sense for them to be aligned?

Reply:
There is no difference, it does the same thing but avoids converting to array, which is an unnecessary operation in this case

Reply:
Should the other `withSources()` be modified to use this approach then?

Reply:
I don't think it makes sense to change the other `withSources` method implementation-wise. There is no need to convert an array into a Collection (just as it doesn't make sense to convert a Collection into an array) just to iterate and add to an internal Set :)

Reply:
It might be because it's late, but I'm not understanding what you're getting at, possibly because I'm explaining it badly.

The method you've added here:

```
    public SmallRyeConfigBuilder withSources(Collection<ConfigSource> configSources) {
        sources.addAll(configSources);
        return this;
    }
```

is different from the existing `withSources()`:
```
    public SmallRyeConfigBuilder withSources(ConfigSource... configSources) {
        Collections.addAll(sources, configSources);
        return this;
    }
```

My question was whether the two methods should have the same implementation?

Reply:
The answer is no, because although they are functionally equivalent, the parameters are different

Reply:
Thanks for your patience, clearly it's time for bed!